### PR TITLE
MNT: fix api changes link in PR template

### PR DESCRIPTION
**Pull request by story645**
## PR summary

As @QuLogic pointed out in https://github.com/matplotlib/matplotlib/pull/27310#issuecomment-1887972417, the pr template now links to the wrong section so changed that. Also tried making the "go to API changes doc" clearer , but don't love the use of admonitions here. 

## PR checklist

- [ ] "closes #0000" is in the body of the PR description to [link the related issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)
- [ ] new and changed code is [tested](https://matplotlib.org/devdocs/devel/testing.html)
- [ ] *Plotting related* features are demonstrated in an [example](https://matplotlib.org/devdocs/devel/document.html#write-examples-and-tutorials)
- [ ] *New Features* and *API Changes* are noted with a [directive and release note](https://matplotlib.org/devdocs/devel/coding_guide.html#new-features-and-api-changes)
- [ ] Documentation complies with [general](https://matplotlib.org/devdocs/devel/document.html#write-rest-pages) and [docstring](https://matplotlib.org/devdocs/devel/document.html#write-docstrings) guidelines

### CODE REVIEW

**Comment by QuLogic** [comment]
I believe this should have been labelled with `Documentation: devdocs` but wasn't due to #27642.

**Comment by QuLogic** [comment]
Tried to re-run it, but it's still complaining about the same stuff, so I guess it's pulling from the base commit, and not top of `main`. Maybe if you rebase this one it'll figure itself out?

**Comment by QuLogic** [comment]
Looks like that one is another minor issue in the config: #27644

**Review by dstansby — APPROVED** [review]

**Comment by timhoffm** [comment]
> @dstansby @QuLogic do either of you have a preference for admonition vs card for the API changes call out?

To me, card feels a bit out of place. Either we regard "API changes" a proper part of the coding guidelines, in which case we should keep the original heading (optionally move that section further down). Or we regard it as a cross-ref hint (think: "if you contribute code, please consider our API changes policy as well"), in which case, I'd go with `.. note::`.

File: doc/devel/coding_guide.rst [hunk 1]
```diff
@@ -7,12 +7,12 @@ Coding guidelines
...
 
-API changes
-===========
 
-If you are adding new features, changing behavior or function signatures, or
-removing classes, functions, methods, or properties, please see the :ref:`api_changes`
-guide.
+.. admonition:: API changes
+
+  If you are adding new features, changing behavior or function signatures, or removing
```
Comment (timhoffm) [inline doc/devel/coding_guide.rst:13]: Semi-OT: new features are not *API changes*. The content of that section is not relevant for new features. I propose to remove "new features" here.

On a side note, new features means new API, which should be in line with our API policy. Unfortunately, we haven't written down anything for that, so can't link anywhere. Informally: Use general best practices and be consistent with our current API. - Maybe that's something to write out more explicitly later. OTOH, I'm unclear how helpful an explicit document on that would be for contributors. We have to review new API anyway. Would PRs become much better API-wise if we had such a document? I doubt it. So IMHO for now, API design guidelines are not worth the effort - both for us writing and for contributors reading.

Comment (timhoffm) [reply doc/devel/coding_guide.rst:13]: Oh! :) In that case I suggest to change

- Page title: "API changes" -> "API guidelines" (in analogy to "Coding guidelines")
- "Announce changes, deprecations, and new features" -> "Announce new features, changes and deprecations", i.e. changed order:  New features are more common (at least for contributions of non-core devs) and thus should be mentioned first.
- Move the section "Adding new features" before "Deprecation" - with the same argument as above.

But that should be a separate PR.

**Review by timhoffm — APPROVED** [review]

**Pull request merged**